pkg/background/update_request_controller.go

```diff
@@ -53,6 +53,17 @@ type controller struct {
 	nsLister   corev1listers.NamespaceLister
 	podLister  corev1listers.PodLister
 
+	// cpolSynced returns true if the cluster policy shared informer has synced at least once
+	cpolSynced cache.InformerSynced
```

Comment:
can we declare this as a list of `cache.InformerSynced` instead?

Reply:
The goal would be to reduce the changes required, when a new Informer is added. Ideally we only update the list when initializing the controller. The rest of the methods would not need to change each time.

Reply:
How would a list work here? Basically, one informer/lister needs a `cache.InformerSynced` which is used to wait for the cache to be synced, they have to set it explicitly when added a new informer:

https://github.com/kyverno/kyverno/blob/53a7720ae9e3c23fd5ed17b03d6b72d98945c92b/pkg/background/generate/cleanup/controller.go#L88-L91

Reply:
Here's an example from Kuberentes, the statefulset controller starts until the pod, statefulset, pvc and controllerrevisions caches are synced. You can see it sets each `cache.InformerSynced` separately.
https://github.com/kubernetes/kubernetes/blob/f4abde9e57cc30efcc8f79c9ebd791d15f4164c4/pkg/controller/statefulset/stateful_set.go#L63-L74

https://github.com/kubernetes/kubernetes/blob/f4abde9e57cc30efcc8f79c9ebd791d15f4164c4/pkg/controller/statefulset/stateful_set.go#L119-L120

https://github.com/kubernetes/kubernetes/blob/f4abde9e57cc30efcc8f79c9ebd791d15f4164c4/pkg/controller/statefulset/stateful_set.go#L136-L137

https://github.com/kubernetes/kubernetes/blob/f4abde9e57cc30efcc8f79c9ebd791d15f4164c4/pkg/controller/statefulset/stateful_set.go#L103-L104

Reply:
Any comment @JimBugwadia ?

Reply:
@realshuting - instead of storing each as a separate field, simply declare a list `SharedInformer.HasSynced`, add to it, and then pass it to `WaitForNamedCacheSync`.

Reply:
@JimBugwadia - I have updated it to a list.

Changing it to a list reduces some work for the developers, but they need to make sure to add `HasSynced` to the `[]cache.InformerSynced` list in order to sync the cache.